ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpTreeBuilderBase.fs

```diff
@@ -770,8 +770,12 @@ type FSharpTreeBuilderBase(lexer, document: IDocument, lifetime, path: VirtualFi
                 x.Done(range, mark, ElementType.ANON_RECORD_FIELD)
             x.Done(range, mark, ElementType.ANON_RECORD_TYPE_USAGE)
 
-        | SynType.StaticConstantNamed(synType1, synType2, _) ->
-            x.MarkTypes(synType1, synType2, range, ElementType.NAMED_STATIC_CONSTANT_TYPE_USAGE)
+        | SynType.StaticConstantNamed(synType1, synType2, constRange) ->
+            // StaticConstantNamed can be dummy, if the corresponding type parameter is empty
+            // i.e. T<int, {caret}>
+            // The real StaticConstantNamed cannot have a length of 1
+            if (getDocumentRange document constRange).Length > 1 then
```

Comment:
This check may fail in some rare recovery cases, e.g. when the range contains some subsequent tokens. I'd suggest checking that ranges are equal or one is inside another, e.g. `Range.equals synType1.Range constRange`, as can show that two unrelated things have the same range: it's never possible in real parsed code and is only used in recovery and various conversions.

Also, please add parser tests for the cases where this fails now. Thanks!